File: easy/sqrt(x)/sqrt(x).cpp
class Solution {
public:
    int mySqrt(int x) {
        long long l = 0;
        long long r = x;
        while (l <= r) {
            long long m = l + (r-l) / 2;
            if (m*m == x || (m*m < x && (m+1)*(m+1) > x)) {
                return m;
            }
            if (m*m < x) {
                l = m + 1;
            }
            if (m*m > x) {
                r = m - 1;
            }
        }
        return 1;
    }
}; 

File: easy/sqrt(x)/sqrt(x).py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        left_bound = 0
        right_bound = x
        while left_bound <= right_bound:
            midpoint = (left_bound + right_bound) // 2
            if midpoint*midpoint == x:
                return midpoint
            if midpoint*midpoint < x and (midpoint+1) * (midpoint+1) > x:
                return midpoint
            if midpoint*midpoint < x:
                left_bound = midpoint + 1
            if midpoint*midpoint > x:
                right_bound = midpoint - 1
